clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -1667,6 +1669,7 @@ private void updateLastSeenEpochIfNewer(TopicPartition topicPartition, OffsetAnd
     public boolean updateAssignmentMetadataIfNeeded(Timer timer) {
         maybeThrowFencedInstanceException();
         maybeInvokeCommitCallbacks();
+        maybeUpdateSubscriptionMetadata();
         backgroundEventProcessor.process();
 
         // Keeping this updateAssignmentMetadataIfNeeded wrapping up the updateFetchPositions as
```

Comment:
Since we're here, and adding logic that adds to the purpose of this `updateAssignmentMetadataIfNeeded`, we could clean up and remove this outdated comment.